addon/services/ajax.js

```diff
@@ -113,9 +113,9 @@ export default Ember.Service.extend({
         );
 
         if (response instanceof AjaxError) {
-          reject(response);
+          Ember.run.join(null, reject, response);
```

Comment:
On another note though, why are we making this change. `Ember.RSVP` promises are exempt from the `auto-run` assertion, to specifically allow for this sort of case, and prevent introducing accidental ordering differences.